lib/zlib.js

```diff
@@ -574,6 +574,13 @@ Zlib.prototype._processChunk = function(chunk, flushFlag, cb) {
   req.callback = callback;
 
   function callback(availInAfter, availOutAfter) {
+    if (this) {
+      // Don't hold onto the buffer and the callback
+      // after they have been used.
+      delete this.buffer;
+      delete this.callback;
```

Comment:
For efficiency reasons, it's better to set the values to null or undefined.  The `delete` operator changes the shape (a.k.a. the hidden class) of the object.